qiskit/circuit/quantumcircuit.py

```diff
@@ -1646,31 +1644,28 @@ def qasm(
 
         string_temp = self.header + "\n"
         string_temp += self.extension_lib + "\n"
-        for register in self.qregs:
-            string_temp += register.qasm() + "\n"
-        for register in self.cregs:
-            string_temp += register.qasm() + "\n"
-
-        bit_labels = {
-            bit: "%s[%d]" % (reg.name, idx)
-            for reg in self.qregs + self.cregs
-            for (idx, bit) in enumerate(reg)
-        }
-
-        regless_qubits = set(self.qubits) - {bit for reg in self.qregs for bit in reg}
-        regless_clbits = set(self.clbits) - {bit for reg in self.cregs for bit in reg}
 
+        regless_qubits = [bit for bit in self.qubits if not self.find_bit(bit).registers]
+        regless_clbits = [bit for bit in self.clbits if not self.find_bit(bit).registers]
+        dummy_registers = []
         if regless_qubits:
-            register_name = self._unique_register_name("qregless_")
-            string_temp += f"qreg {register_name}[{len(regless_qubits)}];\n"
-            bit_labels.update(
-                {bit: f"{register_name}[{idx}]" for idx, bit in enumerate(regless_qubits)}
-            )
+            dummy_registers.append(QuantumRegister(name="qregless", bits=regless_qubits))
```

Comment:
We have some amount of namespace-collision handling already to avoid these cases:
```python
In [1]: from qiskit.circuit import QuantumCircuit, QuantumRegister, Qubit
   ...: print(QuantumCircuit(QuantumRegister(2, "qregless"), [Qubit()]).qasm())
OPENQASM 2.0;
include "qelib1.inc";
qreg qregless[2];
qreg qregless0[1];
```